sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/sql/ParsedStatement.scala

```diff
@@ -36,8 +38,11 @@ import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan
 private[sql] abstract class ParsedStatement extends LogicalPlan {
   // Redact properties and options when parsed nodes are used by generic methods like toString
   override def productIterator: Iterator[Any] = super.productIterator.map {
```

Comment:
why don't we add like:

```scala
protected def options: Map[String, String] = { Map.empty }
protected def properties: Map[String, String] = { Map.empty }
```

and, 

```diff
-    options: Map[String, String],
+    override val options: Map[String, String],
```

at implementation of this classes? Seems like currently we'll check every maps whatever it is.

Reply:
This is not flexible enough to automatically redact other field names. Please check out the solution in my new commit.

Reply:
We can just redact `options` and `properties` and remove `productIterator` override. Wasn't that the purpose of doing this?

I am currently unable to follow why we need to it for every map argument in implemented case classes. How about array?

Reply:
My guess the intention is to auto-redact sensitive info for any Map field in subclasses. More subclasses will be added and sensitive fields may not be called `options` or `properties`.